src/stream.rs

```diff
@@ -45,7 +45,18 @@ pub struct ChannelDescriptor {
 }
 
 impl ChannelDescriptor {
-    pub fn new(
+    /// Creates a new channel descriptor from the given `name`.
+    ///
+    /// If you would like to include tags, see also [`Self::new_with_tags`].
+    pub fn new(name: impl Into<String>) -> Self {
+        Self {
+            name: name.into(),
+            tags: BTreeMap::new(),
```

Comment:
on second thought - lets switch this to Option<BTreeMap> to avoid allocating an empty one if it's not needed?